### PR TITLE
[SPARK-39381][SQL] Make vectorized orc columar writer batch size configurable

### PR DESCRIPTION
**Pull request by cxzl25**
### What changes were proposed in this pull request?
Introduce configuration items and set batch size when constructing orc writer.

### Why are the changes needed?
Now vectorized columar orc writer batch size is default 1024.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
add UT

### CODE REVIEW

**Review by dongjoon-hyun — CHANGES_REQUESTED** [review]
Thank you for making a PR, @cxzl25 . 
Could you make a test case for your contribution, please?

File: sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/orc/OrcQuerySuite.scala [hunk 1]
```diff
@@ -844,6 +844,21 @@ abstract class OrcQuerySuite extends OrcQueryTest with SharedSparkSession {
       df.write.format("orc").save(path)
     }
   }
+
+  test("SPARK-39387: Make vectorized orc columar writer batch size configurable") {
+    Seq(10, 100).foreach(batchSize => {
+      withSQLConf(SQLConf.ORC_VECTORIZED_WRITER_BATCH_SIZE.key -> batchSize.toString) {
```
Comment (cxzl25) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/orc/OrcQuerySuite.scala:850]: If the default value of 1024 is used here, although intercet Throwable, OOM causes the spark process to exit abnormally, and the test will not succeed.

File: sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/orc/OrcQuerySuite.scala [hunk 2]
```diff
@@ -844,6 +844,21 @@ abstract class OrcQuerySuite extends OrcQueryTest with SharedSparkSession {
       df.write.format("orc").save(path)
     }
   }
+
+  test("SPARK-39387: Make vectorized orc columar writer batch size configurable") {
```
Comment (dongjoon-hyun) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/orc/OrcQuerySuite.scala:848]: ID looks wrong to me. Could you double-check? I expected SPARK-39381.

Comment (cxzl25) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/orc/OrcQuerySuite.scala]: Thanks, i fixed it now.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you, @cxzl25 .
Merged to master.

**Pull request closed without merging**